common/src/util/og.ts

```diff
@@ -0,0 +1,17 @@
+import { DOMAIN } from 'common/envs/constants'
+import { removeUndefinedProps } from 'common/util/object'
+
+export function buildOgUrl<P extends Record<string, string | undefined>>(
+  props: P,
+  endpoint: string
+) {
+  const generateUrlParams = (params: P) =>
+    new URLSearchParams(removeUndefinedProps(params) as any).toString()
```

Comment:
I don't really think it makes sense for this function to take `foo: undefined` parameter pairs and then throw them directly in the trash. I would have filtered those out outside of this function.